staging/src/k8s.io/kubectl/pkg/cmd/delete/delete_test.go

```diff
@@ -263,6 +264,53 @@ func TestDeleteObject(t *testing.T) {
 	}
 }
 
+func TestDeleteObjectWithInteractive(t *testing.T) {
+	cmdtesting.InitTestErrorHandler(t)
+	_, _, rc := cmdtesting.TestData()
+
+	tf := cmdtesting.NewTestFactory().WithNamespace("test")
+	defer tf.Cleanup()
+
+	codec := scheme.Codecs.LegacyCodec(scheme.Scheme.PrioritizedVersionsAllGroups()...)
+
+	tf.UnstructuredClient = &fake.RESTClient{
+		NegotiatedSerializer: resource.UnstructuredPlusDefaultContentConfig().NegotiatedSerializer,
+		Client: fake.CreateHTTPClient(func(req *http.Request) (*http.Response, error) {
+			switch p, m := req.URL.Path, req.Method; {
+			case p == "/namespaces/test/replicationcontrollers/redis-master" && m == "DELETE":
+				return &http.Response{StatusCode: http.StatusOK, Header: cmdtesting.DefaultHeader(), Body: cmdtesting.ObjBody(codec, &rc.Items[0])}, nil
+			default:
+				t.Fatalf("unexpected request: %#v\n%#v", req.URL, req)
+				return nil, nil
+			}
+		}),
+	}
+
+	streams, in, buf, _ := genericclioptions.NewTestIOStreams()
+	fmt.Fprint(in, "y")
+	cmd := NewCmdDelete(tf, streams)
+	cmd.Flags().Set("filename", "../../../testdata/redis-master-controller.yaml")
+	cmd.Flags().Set("output", "name")
+	cmd.Flags().Set("interactive", "true")
+	cmd.Run(cmd, []string{})
+
+	if buf.String() != "You are about to delete the following 1 resource(s):\nReplicationController/redis-master\nDo you want to continue? (y/n): replicationcontroller/redis-master\n" {
+		t.Errorf("unexpected output: %s", buf.String())
+	}
+
+	streams, in, buf, _ = genericclioptions.NewTestIOStreams()
+	fmt.Fprint(in, "n")
+	cmd = NewCmdDelete(tf, streams)
+	cmd.Flags().Set("filename", "../../../testdata/redis-master-controller.yaml")
+	cmd.Flags().Set("output", "name")
+	cmd.Flags().Set("interactive", "true")
+	cmd.Run(cmd, []string{})
+
+	if buf.String() != "You are about to delete the following 1 resource(s):\nReplicationController/redis-master\nDo you want to continue? (y/n): deletion is cancelled\n" {
```

Comment:
Besides asserting that the correct message is printed as you have here, can we also assert that the expected request was (or in this case, wasn't) made?